charts/k8s-daemonset/templates/_daemonset_spec.tpl

```diff
@@ -0,0 +1,427 @@
+
+                    containerPorts:
+                      app:
+                        port: 8080
+                        protocol: TCP
+
+              Then this is merged with the default and becomes:
+
+                    containerPorts:
+                      app:
+                        port: 8080
+                        protocol: TCP
+                      http:
+                        port: 80
+                        protocol: TCP
+                      https:
+                        port: 443
+                        protocol: TCP
+
+              and so it becomes append as opposed to replace. To handle this,
+              we allow users to explicitly disable predefined ports. So if the user wants to
+              replace the ports with their own, they would provide the following values file:
+
+                    containerPorts:
+                      app:
+                        port: 8080
+                        protocol: TCP
+                      http:
+                        disabled: true
+                      https:
+                        disabled: true
+            */ -}}
+            {{- range $key, $portSpec := .Values.containerPorts }}
+            {{- if not $portSpec.disabled }}
+            - name: {{ $key }}
+              containerPort: {{ int $portSpec.port }}
+              protocol: {{ $portSpec.protocol }}
+            {{- end }}
+            {{- end }}
+          {{- end }}
+
+
+          {{- if .Values.securityContext }}
+          securityContext:
+{{ toYaml .Values.securityContext | indent 12 }}
+          {{- end}}
+          resources:
+{{ toYaml .Values.containerResources | indent 12 }}
+
+          {{- if or .Values.lifecycleHooks.enabled (gt (int .Values.shutdownDelay) 0) }}
+          lifecycle:
+            {{- if and .Values.lifecycleHooks.enabled .Values.lifecycleHooks.postStart }}
+            postStart:
+{{ toYaml .Values.lifecycleHooks.postStart | indent 14 }}
+            {{- end }}
+
+            {{- if and .Values.lifecycleHooks.enabled .Values.lifecycleHooks.preStop }}
+            preStop:
+{{ toYaml .Values.lifecycleHooks.preStop | indent 14 }}
+            {{- else if gt (int .Values.shutdownDelay) 0 }}
+            # Include a preStop hook with a shutdown delay for eventual consistency reasons.
+            # See https://blog.gruntwork.io/delaying-shutdown-to-wait-for-pod-deletion-propagation-445f779a8304
+            preStop:
+              exec:
+                command:
+                  - sleep
+                  - "{{ int .Values.shutdownDelay }}"
+            {{- end }}
+
+          {{- end }}
+
+          {{- /* START ENV VAR LOGIC */ -}}
+          {{- if index $hasInjectionTypes "hasEnvVars" }}
+          env:          
+            - name:  SYSLOG_HOST
+              value: "sysloghost"
+            - name:  SYSLOG_PORT
+              value: "514"
+            - name:  SYSLOG_PROTOCOL
+              value: "udp"         
```

Comment:
These appear to be specific to the fluentd example, and should instead be managed in the `values.yaml` for that example instead of hardcoded here.